assets/javascripts/running.js

```diff
@@ -118,9 +118,9 @@ function updateTestStatus(newStatus) {
             updateModuleSelection($(moduleSelect).find('option'), developerMode.currentModuleIndex);
         }
 
-        // handle case when results table has not been created yet
+        // handle case when results table has not been created yet or no modules are present yet
         const resultsTable = document.getElementById('results');
-        if (!resultsTable) {
+        if (!resultsTable || !Array.from(resultsTable.tBodies).find(tbody => tbody.rows.length > 0)) {
```

Comment:
As per usual when something was manually tested, and ends up being revisited in a follow-up PR I need to ask... can we unit test this?

Reply:
I know. Unfortunately there are no UI-only tests for this JavaScript code and adding checks for this within the fullstack test would be hard to make stable and debug. So I would need to refactor the JavaScript code to make it testable and add a whole new test with mocking and JavaScript injection similar to `t/ui/25-developer_mode.t`. Maybe we should invest the effort at some point but I wouldn't do it just for the sake of this change.

Reply:
I think we improved the full stack test enough to be able to add a simple check for this assuming that the prerequisites are properly setup in the full stack test. Could you please give it a try?

Reply:
I could add a check for test modules being shown but it will be hard to make a scenario where this condition is guaranteed to be relevant. Hence I'd rather go for an UI-only test.

Reply:
Are you sure it is hard? Because we already use `PAUSE_AT` in t/full-stack.t . Is that not guaranteeing that a job is consistently in "running" and you have all the time you need to check something?

But considering alternatives: I see in t/ui/01-list.t that we create "running" modules on the fly, and in https://github.com/os-autoinst/openQA/blob/master/t/ui/18-tests-details.t#L237 and following we check some details on a "running" job. Can the later not be extended to first reproduce the problem and then verify your fix?

Reply:
> Are you sure it is hard? Because we already use PAUSE_AT in t/full-stack.t . Is that not guaranteeing that a job is consistently in "running" and you have all the time you need to check something?

That's not the problem. However, the setup needed to be that we're browsing the details page while the job is still scheduled and only start the job then. That would be possible (maybe that's already how the test works). But it still leaves room for one race condition: The test modules might be computed quite fast. Since the JavaScript-side relies on polling it might not get into the state when there's an empty job details table and therefore the condition is not tested.

The alternative sounds actually like a good idea. Maybe I don't have to come up with a whole new test setup. Or I'll just extend the developer mode UI-only test (maybe renaming it to `25-live.t`).

Reply:
I've found a fitting place within `ui/18-test-details.t`. It wasn't so bad after all because I simply changed the state of the job modules using some DBIx calls. That's really a much better solution than adding it to the fullstack test.